src/main/scala/mesosphere/marathon/api/v2/AppTasksResource.scala

```diff
@@ -41,30 +45,32 @@ class AppTasksResource @Inject() (
   @GET
   def indexJson(
     @PathParam("appId") id: String,
-    @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
-    val tasksResponse = async {
+    @Context req: HttpServletRequest, @Suspended asyncResponse: AsyncResponse): Unit = sendResponse(asyncResponse) {
+    async {
+      implicit val identity = await(authenticatedAsync(req))
       val instancesBySpec = await(instanceTracker.instancesBySpec)
       id match {
         case GroupTasks(gid) =>
           val groupPath = gid.toRootPath
           val maybeGroup = groupManager.group(groupPath)
           withAuthorization(ViewGroup, maybeGroup, unknownGroup(groupPath)) { group =>
-            ok(jsonObjString("tasks" -> runningTasks(group.transitiveAppIds, instancesBySpec).toRaml))
+            val tasks = await(runningTasks(group.transitiveAppIds, instancesBySpec)).toRaml
```

Comment:
This is not compiling yet since we cannot have `await` in the closure. @timcharper, would you have an alternative for `withAuthorization`?